pkg/operator/ceph/object/bucket/provisioner.go

```diff
@@ -136,29 +138,27 @@ func (p Provisioner) Grant(options *apibkt.BucketOptions) (*bktv1alpha1.ObjectBu
 		return nil, errors.Wrapf(err, "bucket %s does not exist", p.bucketName)
 	}
 
-	p.accessKeyID, p.secretAccessKey, err = p.createCephUser("")
+	// get or create ceph user
+	p.accessKeyID, p.secretAccessKey, err = p.createCephUser(options.UserID)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrap(err, "Provision: can't create ceph user")
 	}
 
-	// need to quota into -1 for restricting creation of new buckets in rgw
-	restrictBucketCreation := -1
+	// restrict creation of new buckets in rgw
+	restrictBucketCreation := 0
```

Comment:
Changing the quota from -1 to 0 is effectively still the same, right? Or was there some side effect of -1?

Reply:
It's actually a behavior fix I noticed. -1 actually disables the quota. 0 is the value that specifies "no new buckets". The commit that added this was pretty old, so it's possible there was some confusion at the time.